homeassistant/components/ipp/sensor.py

```diff
@@ -0,0 +1,178 @@
+        enabled_default: bool = True,
+        entry_id: str,
+        icon: str,
+        key: str,
+        name: str,
+        unit_of_measurement: Optional[str] = None,
+    ) -> None:
+        """Initialize IPP sensor."""
+        self._unit_of_measurement = unit_of_measurement
+        self._key = key
+
+        super().__init__(
+            entry_id=entry_id,
+            coordinator=coordinator,
+            name=name,
+            icon=icon,
+            enabled_default=enabled_default,
+        )
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return f"{self.coordinator.data.info.uuid}_{self._key}"
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit this state is expressed in."""
+        return self._unit_of_measurement
+
+
+class IPPMarkerSensor(IPPSensor):
+    """Defines an IPP marker sensor."""
+
+    def __init__(
+        self, entry_id: str, coordinator: IPPDataUpdateCoordinator, marker_index: int
+    ) -> None:
+        """Initialize IPP marker sensor."""
+        self.marker_index = marker_index
+
+        super().__init__(
+            coordinator=coordinator,
+            entry_id=entry_id,
+            icon="mdi:water",
+            key=f"marker_{marker_index}",
+            name=f"{coordinator.data.info.name} {coordinator.data.markers[marker_index].name}",
+            unit_of_measurement=UNIT_PERCENTAGE,
+        )
+
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, Any]]:
+        """Return the state attributes of the entity."""
+        return {
+            ATTR_MARKER_HIGH_LEVEL: self.coordinator.data.markers[
+                self.marker_index
+            ].high_level,
+            ATTR_MARKER_LOW_LEVEL: self.coordinator.data.markers[
+                self.marker_index
+            ].low_level,
+            ATTR_MARKER_TYPE: self.coordinator.data.markers[
+                self.marker_index
+            ].marker_type,
+        }
+
+    @property
+    def state(self) -> Union[None, str, int, float]:
+        """Return the state of the sensor."""
+        return self.coordinator.data.markers[self.marker_index].level
+
+
+class IPPPrinterSensor(IPPSensor):
+    """Defines an IPP printer sensor."""
+
+    def __init__(self, entry_id: str, coordinator: IPPDataUpdateCoordinator) -> None:
+        """Initialize IPP printer sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            entry_id=entry_id,
+            icon="mdi:printer",
+            key="printer",
+            name=coordinator.data.info.name,
+            unit_of_measurement=None,
+        )
+
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, Any]]:
+        """Return the state attributes of the entity."""
+        return {
+            ATTR_INFO: self.coordinator.data.info.printer_info,
+            ATTR_SERIAL: self.coordinator.data.info.serial,
```

Comment:
Ok to keep it here for now. I think that I want to start adding this to the device registry so it will be easier for people to find this info for their devices.

Reply:
so would that be something core support for first or how would I get conditionally provide that?

Reply:
We first need core support.